cmd/gke-gcloud-auth-plugin/gcloud_edge_cloud_token_provider.go

```diff
@@ -37,11 +38,16 @@ func (p *gcloudEdgeCloudTokenProvider) token() (string, *time.Time, error) {
 
 func (p *gcloudEdgeCloudTokenProvider) useCache() bool { return true }
 
-func getGcloudEdgeCloudTokenRaw(project string, location string, clusterName string, impersonateServiceAccount string) ([]byte, error) {
-	args := []string{"edge-cloud", "container", "clusters", "print-access-token", clusterName, fmt.Sprintf("--project=%s", project), fmt.Sprintf("--location=%s", location), "--format=json"}
-	if impersonateServiceAccount != "" {
-		args = append(args, fmt.Sprintf("--impersonate-service-account=%s", impersonateServiceAccount))
+func (p *gcloudEdgeCloudTokenProvider) getExtraArgs() []string {
+	args := []string{p.clusterName, fmt.Sprintf("--project=%s", p.project), fmt.Sprintf("--location=%s", p.location), "--format=json"}
+	if p.impersonateServiceAccount != "" {
+		args = append(args, fmt.Sprintf("--impersonate-service-account=%s", p.impersonateServiceAccount))
 	}
+	return args
+}
 
+func getGcloudEdgeCloudTokenRaw(extraArgs ...string) ([]byte, error) {
+	args := []string{"edge-cloud", "container", "clusters", "print-access-token"}
+	args = append(args, extraArgs...)
 	return executeCommand("gcloud", args...)
```

Comment:
It would be great if we could parameterize executeCommand as p.executeCommand and add unit test to ensure commands are getting created correctly when it reaches execute command.